src/lib/plugins/live-column/live-column.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit, Input, ChangeDetectionStrategy, NgZone } from '@angular/core';
+import { PluginService } from '../plugin.service';
+import { Fastdom } from '../../../core/services/fastdom';
+import { GRID_PREFIX } from 'ng2-qgrid/core/definition';
+
+@Component({
+	selector: 'q-grid-live-column',
+	template: '<ng-content></ng-content>',
+	providers: [PluginService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveColumnComponent implements OnInit {
+	private startPos: number;
+	private endPos: number;
+	@Input('duration') duration = 200;
+
+	constructor(private plugin: PluginService, private zone: NgZone) { }
+
+	ngOnInit() {
+		const { model } = this.plugin;
+		let currentColumns: any[];
+
+		model.animation({
+			apply: model.animation().apply.concat((memo, context, complete) => {
+				const previousColumns = currentColumns;
+				currentColumns = memo.columns ? memo.columns[0] : currentColumns;
+
+				if (!previousColumns || !memo.columns) {
+					complete();
+					return;
+				}
+
+				const id = model.data().id.column;
+				const animations = [];
+
+				this.startPos = currentColumns.length;
+				this.endPos = 0;
+
+				for (let colId = 0, length = previousColumns.length; colId < length; colId++) {
```

Comment:
I'm not sure what next `for` loops are doing, could you describe it in code more friendly please:)